### PR TITLE
Refactor authorization middleware for readability

### PR DESCRIPTION
**Pull request by rwz**

### CODE REVIEW

**Comment by rwz** [comment]
This should be merged after the https://github.com/heroku/telex/pull/68 because as it is now, Codeclimate considers 2.0+ syntax unparsable.

**Comment by elight** [comment]
Codeclimate failures

File: lib/middleware/user_authenticator.rb [hunk 1]
```diff
@@ -5,38 +5,43 @@ def initialize(app)
...
+      auth.provided? && auth.basic? && auth.credentials&.last
+    end
+
+    def skip_auth?(env)
+      env["PATH_INFO"]&.end_with?("/read.png")
     end
 
     def lookup_user(key:)
-      return nil unless key =~ Pliny::Middleware::RequestID::UUID_PATTERN
+      return nil unless Pliny::Middleware::RequestID::UUID_PATTERN === key
```
Comment (dmcinnes) [inline lib/middleware/user_authenticator.rb]: TIL you can `===` a Regexp and a String, makes sense.

**Comment by dmcinnes** [comment]
Looks great! WAY cleaner. Remove the superfluous `private` Codeclimate found and we're good.

**Comment by rwz** [comment]
@dmcinnes done :)

**Comment by dmcinnes** [comment]
Thanks! 👍

**Pull request merged**